features/all-positions/LiquidationActionDialog.tsx

```diff
@@ -0,0 +1,488 @@
+    liquidations[props.selectedSponsor] &&
+    liquidationLiveness &&
+    priceId &&
+    disputeBondPct &&
+    collAllowance &&
+    collBalance &&
+    collSymbol &&
+    disputeBondPct &&
+    liquidationLiveness &&
+    withdrawalLiveness &&
+    currentTime &&
+    tokenSymbol &&
+    finalFee
+  ) {
+    const liquidatedPosition = liquidations[props.selectedSponsor];
+    const invertDisputablePrice = isPricefeedInvertedFromTokenSymbol(
+      tokenSymbol
+    );
+    const priceIdUtf8 = utils.parseBytes32String(priceId);
+    const collBalanceNum = collBalance || 0;
+    const finalFeeNum = finalFee || 0;
+    const disputeBondNum = Number(fromWei(disputeBondPct)) || 0;
+
+    const needCollateralAllowance = () => {
+      if (collAllowance === "Infinity") return false;
+      return collAllowance < finalFee;
+    };
+
+    const requiredDisputeBond =
+      Number(fromWei(disputeBondPct)) *
+      Number(liquidatedPosition.lockedCollateral);
+    const collRequired = finalFeeNum + requiredDisputeBond;
+    const collBalanceTooLow = collBalanceNum < collRequired;
+
+    const liquidationTimeRemaining =
+      Number(liquidatedPosition.liquidationTimestamp) +
+      Number(withdrawalLiveness) -
+      Number(currentTime);
+
+    const pendingLiquidationTimeString =
+      liquidationTimeRemaining > 0
+        ? Math.max(0, Math.floor(liquidationTimeRemaining / 3600)) +
+          ":" +
+          Math.max(0, Math.floor((liquidationTimeRemaining % 3600) / 60)) +
+          ":" +
+          Math.max(0, (liquidationTimeRemaining % 3600) % 60)
+        : "None";
+
+    const translateLiquidationStatus = (
+      liquidationTimestamp: number,
+      liquidationStatus: string
+    ) => {
+      const liquidationTimeRemaining =
+        liquidationTimestamp +
+        liquidationLiveness.toNumber() -
+        Math.floor(Date.now() / 1000);
+      if (liquidationTimeRemaining > 0 && liquidationStatus === "PreDispute") {
+        return {
+          status: "Liquidation Pending",
+          description:
+            "The liquidation is currently pending for the liveness period. The liquidation can be disputed during this time.",
+        };
+      } else if (
+        liquidationTimeRemaining <= 0 &&
+        liquidationStatus === "PreDispute"
+      ) {
+        return {
+          status: "Liquidation Succeeded",
+          description:
+            "The liquidation passed the liquidation liveness period undisputed.",
+        };
+      } else if (liquidationStatus === "PendingDispute") {
+        return {
+          status: "Dispute Pending",
+          description:
+            "The liquidation was disputed and will be resolved once the DVM returns a resolution price.",
+        };
+      } else if (liquidationStatus === "DisputeSucceeded") {
+        return {
+          status: "Dispute Succeeded",
+          description:
+            "The liquidation was disputed and the DVM deemed the liquidation invalid (valid dispute).",
```

Comment:
is it perhaps clearer to say "the resolved price made the liquidation invalid"? Its a little bit more wordy so I don't think its exactly right, but we shouldn't say that the DVM deemed the liquidation X, rather the DVM resolved a price, and the resolved price made the liquidation X

Reply:
I have updated accordingly. LMK what you think.